gmime/gmime-parser.c

```diff
@@ -922,7 +909,7 @@ header_parse (GMimeParser *parser, GMimeParserOptions *options)
 		priv->header_cb (parser, header->name, header->raw_value,
 				 header->offset, priv->user_data);
 	
-	if (can_warn && (!is_7bit_clean (header->name) || !g_utf8_validate (header->raw_value, -1, NULL)))
+	if (can_warn && (g_mime_utils_text_is_8bit ((const unsigned char *) header->name, strlen (header->name)) || !g_utf8_validate (header->raw_value, -1, NULL)))
```

Comment:
Any reason you chose to use g_mime_utils_text_is_8bit() instead of is_7bit_clean() other than attempting to reduce code duplication?

One problem with the is_8bit() function is that you have to strlen() the field name whereas with is_7bit_clean(), you don't. This is just a performance consideration and most likely minor due to the fact that field names are typically short, but I figured I'd bring it up anyway.

Reply:
Yes, the reason was an attempt to remove duplicated code.  IMO, the additional strlen() will typically not introduce any overhead, as the “usual” case will be can_warn == FALSE, i.e. strlen() is not called.

Reply:
That's fine. I merged that as it is and left it alone. I just wanted to be aware of any problems if there were any :)